jgrapht-core/src/main/java/org/jgrapht/alg/tour/PalmerHamiltonianCycle.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.tour;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.GraphTests;
+import org.jgrapht.alg.interfaces.HamiltonianCycleAlgorithm;
+import org.jgrapht.graph.GraphWalk;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Palmer's algorithm for computing Hamiltonian cycles in graphs that meet Ore's condition.
```

Comment:
Can you add some text on the intuition behind Ore's condition? This makes it easier for users unfamiliar with Ores condition to assess whether this algorithm can be used to solve the HC problem on their graph.